destination/service/server.go

```diff
@@ -139,36 +176,54 @@ func (s *Server) AlterTable(ctx context.Context, in *pb.AlterTableRequest) (*pb.
 }
 
 func (s *Server) Truncate(ctx context.Context, in *pb.TruncateRequest) (*pb.TruncateResponse, error) {
+	var softDeleteColumn *string = nil
+	if in.Soft != nil && in.Soft.DeletedColumn != "" {
+		softDeleteColumn = &in.Soft.DeletedColumn
+	}
+
+	truncateBefore := time.Unix(in.UtcDeleteBefore.Seconds, int64(in.UtcDeleteBefore.Nanos)).UTC()
+	deleteType := "hard"
+	if softDeleteColumn != nil {
+		deleteType = "soft"
+	}
+
+	log.Info(fmt.Sprintf("[Truncate] Starting %s delete for %s.%s, synced_column=%s, truncate_before=%s",
+		deleteType, in.SchemaName, in.TableName, in.SyncedColumn, truncateBefore.Format(time.RFC3339)))
+
 	conn, err := db.GetClickHouseConnection(ctx, in.GetConfiguration())
 	if err != nil {
+		log.Error(fmt.Errorf("[Truncate] GetClickHouseConnection error for %s.%s: %w", in.SchemaName, in.TableName, err))
 		return FailedTruncateTableResponse(in.SchemaName, in.TableName, err), nil
 	}
 	defer conn.Close()
 
+	log.Info(fmt.Sprintf("[Truncate] Checking if table %s.%s exists", in.SchemaName, in.TableName))
 	// should not be failed if the table does not exist, as per SDK documentation
 	tableDescription, err := conn.DescribeTable(ctx, in.SchemaName, in.TableName)
 	if err != nil {
+		log.Error(fmt.Errorf("[Truncate] DescribeTable error for %s.%s: %w", in.SchemaName, in.TableName, err))
 		return FailedTruncateTableResponse(in.SchemaName, in.TableName, err), nil
 	}
 	if tableDescription == nil || len(tableDescription.Columns) == 0 {
+		log.Info(fmt.Sprintf("[Truncate] Table %s.%s not found, skipping truncate", in.SchemaName, in.TableName))
 		return SuccessfulTruncateTableResponse(), nil
 	}
 
-	var softDeleteColumn *string = nil
-	if in.Soft != nil && in.Soft.DeletedColumn != "" {
-		softDeleteColumn = &in.Soft.DeletedColumn
-	}
-
-	truncateBefore := time.Unix(in.UtcDeleteBefore.Seconds, int64(in.UtcDeleteBefore.Nanos)).UTC()
+	log.Info(fmt.Sprintf("[Truncate] Executing %s truncate for %s.%s", deleteType, in.SchemaName, in.TableName))
 	err = conn.TruncateTable(ctx, in.SchemaName, in.TableName, in.SyncedColumn, truncateBefore, softDeleteColumn)
 	if err != nil {
+		log.Error(fmt.Errorf("[Truncate] TruncateTable failed for %s.%s: %w", in.SchemaName, in.TableName, err))
 		return FailedTruncateTableResponse(in.SchemaName, in.TableName, err), nil
 	}
 
+	log.Info(fmt.Sprintf("[Truncate] Completed successfully for %s.%s", in.SchemaName, in.TableName))
 	return SuccessfulTruncateTableResponse(), nil
 }
 
 func (s *Server) WriteHistoryBatch(ctx context.Context, in *pb.WriteHistoryBatchRequest) (*pb.WriteBatchResponse, error) {
+	log.Info(fmt.Sprintf("[WriteHistoryBatch] Starting for %s.%s with earliest_start_files=%d, replace_files=%d, update_files=%d, delete_files=%d",
```

Comment:
1. does Fivetran provide any recommendations about log levels? this one might be more appropriate for `debug`